src/store/helpers/index.js

```diff
@@ -26,14 +26,18 @@ export const createRSQLQuery = (state) => transformToRSQL({
   ])
 })
 
-export const createBiobankRSQLQuery = (state) => transformToRSQL({
-  operator: 'AND',
-  operands: flatten([
-    createInQuery('country', state.showCountryFacet ? state.country.filters : state.preConfiguredCountyCode),
-    createInQuery('id', state.biobank_quality.biobanks),
-    createInQuery('covid19biobank', state.covid19.filters)
-  ])
-})
+export const createBiobankRSQLQuery = (state) => {
+  const filters = transformToRSQL({
+    operator: 'AND',
+    operands: flatten([
+      createInQuery('country', state.showCountryFacet ? state.country.filters : state.preConfiguredCountyCode),
+      createInQuery('id', state.biobank_quality.biobanks)
+    ])
+  })
+  const covidfilter = createANDQuery('covid19biobank', state.covid19.filters)
+
+  if (filters !== '' && covidfilter !== '') { return `${filters};${covidfilter}` } else return filters || covidfilter
```

Comment:
Leave the string manipulation to the rsql library.
I expect you can replace the `createInQuery` with the `createANDQuery` and then leave the transformation to rsql as is.